frontend/socketutil.lua

```diff
@@ -0,0 +1,120 @@
+--[[--
+This module contains miscellaneous helper functions specific to our usage of LuaSocket/LuaSec.
+]]
+
+local Version = require("version")
+local http = require("socket.http")
+local https = require("ssl.https")
+local socket = require("socket")
+
+local socketutil = {
+    -- Init to the default LuaSocket/LuaSec values
+    block_timeout = 60,
+    total_timeout = -1,
+}
+
+--- Builds a sensible UserAgent that fits Wikipedia's UA policy <https://meta.wikimedia.org/wiki/User-Agent_policy>
+local socket_ua = http.USERAGENT
+socketutil.USER_AGENT = "KOReader/" .. Version:getShortVersion() .. " (https://koreader.rocks/) " .. socket_ua:gsub(" ", "/")
+-- Monkey-patch it in LuaSocket, as it already takes care of inserting the appropriate header to its requests.
+http.USERAGENT = socketutil.USER_AGENT
+
+--- Common timeout values
+-- Large content
+socketutil.LARGE_BLOCK_TMOUT = 10
+socketutil.LARGE_TOTAL_TMOUT = 30
+-- File downloads
+socketutil.FILE_BLOCK_TMOUT = 15
+socketutil.FILE_TOTAL_TMOUT = 60
```

Comment:
TMOUT CHTMOUT !
(As that is used in the other modules, it ought to be readable & spellable :)